### ISSUE TITLE
Assignment Submission

### ISSUE THREAD

**Issue by jayathu** [jayathu/NewYorkTimesArticleSearch-Project1#3]
@codepathreview @codepath

Trying to recall and figure out how to link the gif I just created. For now, I have updated the ReadMe with all the details. I will update the file with the link to the gif as soon as I figure it out. Please review the assignment and let me know!

Thanks.

**Comment by nesquena** [comment]
@jayathu see your old project here: https://raw.githubusercontent.com/jayathu/InstagramClient/master/README.md (at the bottom) for how to link the gif (after uploading to imgur)

**Comment by jayathu** [comment]
@nesquena Yeah I've been staring at that page forever now :-D For some reason, I can't figure out how did I upload the gif the last time! I dont seem to find that option on the website now :(

**Comment by nesquena** [comment]
1. Upload GIF to [imgur](http://imgur.com/)
2. Add `<img src='URL' title='Video Walkthrough' width='' alt='Video Walkthrough' />` to README

Make sure the extension is 'gif' and not 'gifv'

**Comment by jayathu** [comment]
I am right there but I only see options to upload images and not gifs.

**Comment by nesquena** [comment]
Is Imgur not allowing you to do the upload? GIFs are images, so they should allow you to upload the GIF like any other image. Keep in mind as a fallback you can also checkin the GIF to this repo directly.

**Comment by jayathu** [comment]
Yes I tried uploading it by clicking on 'Upload Image' and it gave me a url which doesn't work and didn't show any preview. Last time, I remember it gave a preview on the website. For now, I will checkin the gif to the repo.

**Comment by nesquena** [comment]
The URL they give you is a gifv, if you remove the v at the end it will work as expected. For example:

![](http://i.imgur.com/R8se5g1.gifv) doesn't work (`http://i.imgur.com/R8se5g1.gifv`)

but this should ![](http://i.imgur.com/R8se5g1.gif) (`http://i.imgur.com/R8se5g1.gif`)

**Comment by nesquena** [comment]
or GIFs aren't working right which is weird. Worst case, link to the GIF url in the readme :)

**Comment by jayathu** [comment]
I tried uploading the gif by "uploading image" again and this is the url I am getting:
http://imgur.com/XhhCyoM

Doesn't work :( So weird!!

**Comment by nesquena** [comment]
Right clicking on that and getting the URL, I see: http://i.imgur.com/XhhCyoM.gif which does work

**Comment by jayathu** [comment]
Sorry, totally overlooked the fact that I didn't add the extension! It works now. Thanks much!

**Comment by codepathreview** [comment]
Nice work, consider adding a few more optional features. A few notes after checking out the code:
- Properly reused the single base method in search activity to fetch results for both initial load and the pagination
- Nice to see you added the search box to the ActionBar!
- Properly used spinners (with populated values) to allow users to select filters
- Properly sent and received relevant filters using a bundle to pass data between activities
- Good to see you used a filters object to represent the filters to apply to the query
-  Consider adding the ability to [share a news article](http://guides.codepath.com/android/Sharing-Content-with-Intents). This is an important learning for this assignment even though it's optional.
- Consider adding [progress bars](http://guides.codepath.com/android/Handling-ProgressBars#progress-within-actionbar) to your app whenever a network request goes out

Also, here's a more detailed [Project 2 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_2_feedback) which covers the most common issues with this submitted project. Read through the [feedback guide point-by-point](http://courses.codepath.com/snippets/intro_to_android/project_2_feedback) to determine how you could improve your submission.

Let us know if you have any other thoughts or questions about this assignment. The next assignment (Twitter Client) will be especially important since it introduces the majority of the remaining pieces necessary to build a fully functional API client with complex feeds of data and user creation.
